WordPress/gutenberg
# Update template part icons in the library mosaic (grid items)

## What?

Updates the icons in the grid items in the library to indicate if the template part is a header, footer, or uncategorized.

Closes https://github.com/WordPress/gutenberg/issues/51905

## Why?


The icons did not match with the icons in the sidebar.

## How?


The PR adds a condition that uses the `categoryID` to determine which icon to display.

## Testing Instructions

- Activate a block theme that has at least one footer, header, and uncategorized template part.
- Open Appearance > Editor > Library.
- In the menu, click on Headers, Footers, and Uncategorized, respectively.
- Confirm that the icons in the sidebar and the preview ( the mosaic /grid) match.

## Screenshots or screencast 
After:
<img width="932" alt="The header template part in the library grid has a header icon." src="https://github.com/WordPress/gutenberg/assets/7422055/339e3c98-5212-4741-95a8-f6e7f3a3bbe6">
